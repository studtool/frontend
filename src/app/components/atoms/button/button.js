import styled from 'styled-components';
import {colors} from '../style';

export const Button = styled.button`
    color: ${colors.textOnPrimary};
    background: ${colors.primary};
    border: none;
    border-radius: 5px;
    :hover {
        background: ${colors.primaryDark};
    };
    outline: none;
`;
