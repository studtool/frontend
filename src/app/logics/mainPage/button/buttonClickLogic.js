import {Logics} from "../../common/logics";
import { actions } from "../../../common/actionsTypes";

class ButtonClickLogic extends Logics{
    constructor(type){
        super(type); // типы событий, которые данный Store будет обрабатывать
        this.states = [];
        BUTTON_CLICK__STATES = { // мб в общие?
            ERROR: "BUTTON_CLICK__ERROR",
            OK: "BUTTON_CLICK__OK"
        };
        
    }

    handle(data = {}){

        if (!data.type == this.type)
            break;
    
        // здесь работа с логикой (пока что прувит мир)
        this.print(data);

        // ещё одна логика, которая меняет состояние элемента (заглушка, пример)
        this.stubStates();

        // вызов у контроллера изменений 
        // MainPageViewController.change(data, this.states);
    }

    format(element, status, data){
        return {
            element: element,
            status: status,
            data: data
        }
    }

    stubStates(){
        this.states.push(
            format("greeting", BUTTON_CLICK__STATES.OK, "Hello, World!")
        )
    }

    print(data){
        console.log("Printing data: ", data);
    }
}

export default new ButtonClickLogic(actions.helloWorldButtonClicked);
