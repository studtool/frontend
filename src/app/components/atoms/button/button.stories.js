import * as React from 'react';
import {storiesOf} from '@storybook/react';
import {Button} from 'App/components/atoms/button/button';

storiesOf('Button', module)
    .add('button default', () => (
        <Button>Button</Button>
    ));
