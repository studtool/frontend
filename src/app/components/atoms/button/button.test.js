import React from 'react';
import {Button} from './button';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('button test', () => {
    const tree = renderer.create(<Button>test</Button>).toJSON();

    expect(tree).toMatchSnapshot();
});
