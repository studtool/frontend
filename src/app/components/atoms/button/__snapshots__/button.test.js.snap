// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test 1`] = `
<button
  className=""
>
  test
</button>
`;
